lua/entities/cfc_shaped_charge/init.lua

```diff
@@ -0,0 +1,164 @@
+AddCSLuaFile( "cl_init.lua" )
+AddCSLuaFile( "shared.lua" )
+include( "shared.lua" )
+ 
+function ENT:Initialize()  
+
+    local owner = self.bombOwner
+
+    if not IsValid( owner ) then
+        self:Remove()
+        return
+    end
+
+    owner.plantedCharges = owner.plantedCharges or 0
+    owner.plantedCharges = owner.plantedCharges + 1
+
+    self.bombHealth  = GetConVar( "cfc_shaped_charge_chargehealth" ):GetInt()
+    self.bombTimer   = GetConVar( "cfc_shaped_charge_timer" ):GetInt()
+    self.blastDamage = GetConVar( "cfc_shaped_charge_blastdamage" ):GetInt()
+    self.blastRange  = GetConVar( "cfc_shaped_charge_blastrange" ):GetInt()
+    self.traceRange  = GetConVar( "cfc_shaped_charge_tracerange" ):GetInt()
+
+    self:SetModel( "models/weapons/w_c4_planted.mdl" )
+    self:PhysicsInit( SOLID_VPHYSICS )
+    self:SetSolid( SOLID_VPHYSICS )
+    self:DrawShadow( false )
+    self:SetCollisionGroup( COLLISION_GROUP_WEAPON )
+
+    self:PhysWake()
+
+    self:CreateLight()
+
+    self.explodeTime = CurTime() + self.bombTimer
+
+    self:EmitSound( "items/ammocrate_close.wav", 100, 100, 1, CHAN_STATIC )
+    self:EmitSound( "npc/roller/blade_cut.wav", 100, 100, 1, CHAN_STATIC )
+
+    self:SetNWFloat( "bombInitiated", CurTime() )
+
+    spawnTime = CurTime()
+    self:bombVisualsTimer()
+end
+
+function ENT:OnTakeDamage ( dmg )
+    self.bombHealth = self.bombHealth - dmg:GetDamage()
+    if self.bombHealth <= 0 then
+
+        if not IsValid( self ) then return end
+
+        local effectdata = EffectData()
+        effectdata:SetOrigin( self:GetPos() )
+        effectdata:SetMagnitude( 8 )
+        effectdata:SetScale( 1 )
+        effectdata:SetRadius( 16 )
+            
+        util.Effect( "Sparks", effectdata )
+        
+        self:EmitSound( "npc/manhack/bat_away.wav", 100, 100, 1, CHAN_STATIC )
+        self:EmitSound( "doors/vent_open1.wav", 100, 100, 1, CHAN_STATIC )
+
```

Comment:
If someone using this addon wanted something special to happen when the bomb gets shot off (such as doing partial damage to the props via SPD or causing a big explosion that damages nearby players), they'd need a hook to latch onto, so let's give them one via a function.
```suggestion
        
        self:OnDestroyed()
        
```

Reply:
Currently not needed.